### ISSUE TITLE
Using `train_predict` as `identification` and `verification` evaluation

### ISSUE THREAD

**Issue by jpainam** [ahangchen/rank-reid#8]
Hi
I came across this paper [A Discriminatively Learned CNN Embedding for
Person Re-identification](https://arxiv.org/pdf/1611.05666) which made use of `identification` and `verification` for their evaluation. I want to confirm if your function `train_predict` is based on such evaluation if it can be used for `verification` and `identification`
Thank you

**Comment by ahangchen** [comment]
Identification and verification is two output branches of the network in that paper, which is also used in [pair_train.py](https://github.com/ahangchen/rank-reid/blob/master/pretrain/pair_train.py). It's nothing to do with evaluation. Evaluation only cares about Dataset. You need to use different evaluation methods according to dataset instructions, like [Market1501](http://www.liangzheng.com.cn/Project/project_reid.html) and [GRID](personal.ie.cuhk.edu.hk/~ccloy/downloads_qmul_underground_reid.html)

**Comment by jpainam** [comment]
According to the repo of the paper, your repo is the [keras](https://keras.io/) implementation of their paper. 
I want to know if after training using [pair_train.py](https://github.com/ahangchen/rank-reid/blob/master/pretrain/pair_train.py), your evaluation in [eval.py](https://github.com/ahangchen/rank-reid/blob/master/pretrain/eval.py) follows the same evaluation protocol as in `identification and verification` paper. 
Thank for confirming this.
I'm using [Market1501](http://www.liangzheng.com.cn/Project/project_reid.html) dataset too

**Comment by ahangchen** [comment]
Yes, evaluation protocol is the same, but implement detail is different. You can get the same rank accuracy with the same similarity ranking table of probe and gallery images from this two different evaluation methods .

**Comment by jpainam** [comment]
Thanks, that's exactly what I was looking for. Would cite your code and  [paper](https://arxiv.org/abs/1803.07293)  soon

**Comment by jpainam** [comment]
One more thing.  Which modeling tools did you use to design this great looking model `The TFusion model consisting of 4 steps`
Thanks.

**Comment by ahangchen** [comment]
You mean fig1 in the paper? It's designed with power point.

**Comment by jpainam** [comment]
Thanks.
